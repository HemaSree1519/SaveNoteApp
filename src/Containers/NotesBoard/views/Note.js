import {Card, CardBody, CardText, CardTitle} from "reactstrap";
import React, {Component} from "react";

export default class Note extends Component {
    render() {
        return (
            <Card name="card" className="note"
                  onClick={() => {
                      this.props.setEditState(this.props.note);
                  }}>
                <CardBody>
                    <CardTitle name="title" className="card-title">{this.props.note["title"]}</CardTitle>
                    <CardText name="content">{this.props.note["content"]}</CardText>
                </CardBody>
            </Card>
        )
    }
};