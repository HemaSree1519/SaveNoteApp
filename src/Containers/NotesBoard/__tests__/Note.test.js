import React from 'react';
import {shallow} from 'enzyme';
import Note from "../views/Note";

describe('EditNote', () => {
    let note;
    let setEditState;
    let component;
    beforeEach(() => {
        note = {
            "id": 2,
            "title": "test",
            "content": "Content here",
            "email": "test@gmail.com",
            "createdAt": "2019-05-12T13:20:40.735Z",
            "updatedAt": "2019-05-12T13:20:40.735Z"
        };
        setEditState = jest.fn();
        component = shallow(<Note note={note} setEditState={setEditState}/>);
    });
    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });
    it('should call edit state function when clicked on card', () => {
        component.find('Card[name="card"]').simulate('click');
        expect(setEditState).toHaveBeenCalledTimes(1);
    })
});