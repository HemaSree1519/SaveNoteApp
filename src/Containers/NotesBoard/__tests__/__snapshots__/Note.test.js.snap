exports[`EditNote should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    note={
        Object {
            "content": "Content here",
            "createdAt": "2019-05-12T13:20:40.735Z",
            "email": "test@gmail.com",
            "id": 2,
            "title": "test",
            "updatedAt": "2019-05-12T13:20:40.735Z",
          }
    }
    setEditState={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <CardBody
        tag="div">
        <CardTitle
                className="card-title"
                tag="div">
                test
        </CardTitle>
        <CardText
                tag="p">
                Content here
        </CardText>
</CardBody>,
      "name": "card",
      "onClick": [Function],
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardTitle
            className="card-title"
            tag="div">
            test
</CardTitle>,
          <CardText
            tag="p">
            Content here
</CardText>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "test",
            "className": "card-title",
            "tag": "div",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Content here",
            "tag": "p",
          },
          "ref": null,
          "rendered": "Content here",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <CardBody
          tag="div">
          <CardTitle
                    className="card-title"
                    tag="div">
                    test
          </CardTitle>
          <CardText
                    tag="p">
                    Content here
          </CardText>
</CardBody>,
        "name": "card",
        "onClick": [Function],
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardTitle
              className="card-title"
              tag="div">
              test
</CardTitle>,
            <CardText
              tag="p">
              Content here
</CardText>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "test",
              "className": "card-title",
              "tag": "div",
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Content here",
              "tag": "p",
            },
            "ref": null,
            "rendered": "Content here",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
